src/Visualization/Utility/SelectionPolygon.cpp

```diff
@@ -182,6 +200,18 @@ std::shared_ptr<PointCloud> SelectionPolygon::CropPointCloudInPolygon(
     return SelectDownSample(input, CropInPolygon(input.points_, view));
 }
 
+std::shared_ptr<TriangleMesh> SelectionPolygon::CropTriangleMeshInRectangle(
+        const TriangleMesh &input, const ViewControl &view)
+{
+    return SelectDownSample(input, CropInRectangle(input.vertices_, view));
+}
+
+std::shared_ptr<TriangleMesh> SelectionPolygon::CropTriangleMeshInPolygon(
+        const TriangleMesh &input, const ViewControl &view)
+{
+    return SelectDownSample(input, CropInPolygon(input.vertices_, view));
+}
```

Comment:
Can we use template functions?

Reply:
Yes, as long as the implementation gets declared in .cpp files and does not leak into .h.